src/main/java/com/aws/greengrass/clientdevices/auth/configuration/MetricsConfiguration.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.configuration;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.util.Coerce;
+import lombok.Getter;
+
+import java.util.Optional;
+
+/**
+ * Represents the metrics part of the component configuration. Acts as an adapter from the GG topics to the domain.
+ * <p>
+ * |---- configuration
+ * |    |---- metrics:
+ * |          |---- emitMetrics
+ * |          |---- emittingFrequency
+ * </p>
+ */
+
+public final class MetricsConfiguration {
+    public static final String METRICS_TOPIC = "metrics";
+    public static final String ENABLE_METRICS = "enableMetrics";
+    public static final String EMITTING_FREQUENCY = "emittingFrequency";
+    @Getter
+    private Optional<Boolean> enableMetrics;
+    @Getter
+    private Optional<Integer> emittingFrequency;
+
+    private MetricsConfiguration(Optional<Boolean> enableMetrics, Optional<Integer> emittingFrequency) {
+        this.enableMetrics = enableMetrics;
+        this.emittingFrequency = emittingFrequency;
+    }
+
+    /**
+     * Create a MetricsConfiguration from the service configuration.
+     *
+     * @param configurationTopics the configuration key of the service configuration
+     * @return MetricsConfiguration
+     */
+    public static MetricsConfiguration from(Topics configurationTopics) {
+        Topics metricsTopic = configurationTopics.lookupTopics(METRICS_TOPIC);
+
+        return new MetricsConfiguration(getEnableMetricsFlagFromConfiguration(metricsTopic),
+                getEmittingFrequencyFromConfiguration(metricsTopic));
+    }
+
+    /**
+     * Compares 2 Metric Configurations and returns true if there's been a change.
+     *
+     * @param config Metric Configuration
+     * @return true if changed, else false
+     */
+    public boolean hasChanged(MetricsConfiguration config) {
+        return config.getEnableMetrics() != getEnableMetrics()
+                || config.getEmittingFrequency() != getEmittingFrequency();
+    }
+
+    private static Optional<Boolean> getEnableMetricsFlagFromConfiguration(Topics metricsTopic) {
```

Comment:
This is not an appropriate use of Optional. Default to true. Keep that business logic contained within this class, otherwise callers need to decide how to interpret getting `Optional.empty()`

Reply:
More context on `Optional`, from one of the Java language designers: https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555

In this case, you could argue that there is "no result" and that this is therefore appropriate. But configuration needs to have a default value, and it makes sense to encapsulate that here.

Reply:
Makes sense, my line of thinking was that customers have the option to provide a config for metrics so in case they didn't we'd just set it as empty, but simply setting the default as false is easier to follow